### ISSUE TITLE
`_isJSON()` doesn't recognize `application/*+json`.

### ISSUE THREAD

**Issue by quoininc-huudatran** [howardabrams/node-mocks-http#238]
When setting `Content-Type: application/hal+json`, the response returns `false` for `res_isJSON()` call.

I would expect it to return `true`.

**Comment by eugef** [comment]
Hi @quoininc-huudatran, atm `_isJSON()` returns true only when `Content-Type: application/json`

But even if `is-type` library would be used - it will be still `false` because
```
typeIs.is('application/hal+json', 'json') === false
```

You can try it here https://runkit.com/embed/p8q50sjg251o

**Comment by quoininc-huudatran** [comment]
According to documentation of type-is: 

> A suffix such as +json. This can be combined with a wildcard such as */vnd+json or application/*+json. The full mime type will be returned if matched.

So we can do:
```
Boolean(typeIs.is('application/hal+json', 'application/*+json'))
```

**Comment by eugef** [comment]
You are right, please review the fix #243
